GitUI/NBugReports/BugReportInvoker.cs

```diff
@@ -172,6 +183,107 @@ void AddIgnoreOrCloseButton()
             }
         }
 
+        private static void ReportDubiousOwnership(Exception exception)
+        {
+            string error = exception.Message;
+            TaskDialogPage pageSecurity = new()
+            {
+                Icon = TaskDialogIcon.Error,
+                Caption = TranslatedStrings.GitSecurityError,
+                Heading = TranslatedStrings.GitDubiousOwnershipHeader,
+                Text = TranslatedStrings.GitDubiousOwnershipText,
+                AllowCancel = true,
+                SizeToContent = true,
+            };
+            int startIndex = error.IndexOf(_dubiousOwnershipSecurityConfigString);
+            string gitConfigTrustRepoCommand = error[startIndex..];
+            string folderPath = error[(startIndex + _dubiousOwnershipSecurityConfigString.Length + 1)..];
+
+            TaskDialogCommandLinkButton openExplorerButton = new(TranslatedStrings.GitDubiousOwnershipOpenRepositoryFolder, allowCloseDialog: false);
+            openExplorerButton.Click += (_, _) => OsShellUtil.OpenWithFileExplorer(PathUtil.ToNativePath(folderPath));
+            pageSecurity.Buttons.Add(openExplorerButton);
+
+            AddTrustRepoButton(TranslatedStrings.GitDubiousOwnershipTrustRepository, gitConfigTrustRepoCommand);
+            AddTrustAllReposButton(TranslatedStrings.GitDubiousOwnershipTrustAllRepositories);
+
+            TaskDialogButton helpButton = TaskDialogButton.Help;
+            helpButton.Click += (_, _) =>
+            {
+                OsShellUtil.OpenUrlInDefaultBrowser("https://git-scm.com/docs/git-config/#Documentation/git-config.txt-safedirectory");
+            };
+
+            pageSecurity.Buttons.Add(helpButton);
+            pageSecurity.Buttons.Add(TaskDialogButton.Close);
+
+            pageSecurity.Expander = new TaskDialogExpander
+            {
+                CollapsedButtonText = TranslatedStrings.GitDubiousOwnershipSeeGitCommandOutput,
+                ExpandedButtonText = TranslatedStrings.GitDubiousOwnershipHideGitCommandOutput,
+                Position = TaskDialogExpanderPosition.AfterFootnote,
+                Text = error,
+            };
+
+            TaskDialogButton button = TaskDialog.ShowDialog(OwnerFormHandle, pageSecurity);
+            if (button == TaskDialogButton.Cancel || button == TaskDialogButton.Close)
```

Comment:
```suggestion
            if (button is (TaskDialogButton.Cancel or TaskDialogButton.Close))
```

Reply:
I don't think that'd compile. I'm not near a computer to verify.

Reply:
![image](https://user-images.githubusercontent.com/4403806/204120011-525247d3-f76b-4e7e-81ba-d15faf54629b.png)